Slim/Formats/Playlists/M3U.pm

```diff
@@ -285,6 +293,10 @@ sub write {
 			if ($title) {
 				print $output "#EXTINF:$secs,$title\n";
 			}
+
+			if (my $addedFromWork = $item->added_from_work) {
```

Comment:
You'd have to use `$track->added_from_work` rather than `$item->...`. That latter is current line in the playlist, not the track object. That should fix at least one of the errors you posted.